toolkit/tools/imagegen/installutils/installutils.go

```diff
@@ -1829,21 +1829,20 @@ func cleanupRpmDatabase(rootPrefix string) (err error) {
 	return
 }
 
-func RunPreInstallScripts(config configuration.SystemConfig) (err error) {
+func RunPreInstallScripts(config configuration.Config) (err error) {
 	const squashErrors = false
 
-	for _, script := range config.PreInstallScripts {
-		ReportActionf("Running pre-install script: %s", path.Base(script.Path))
-		logger.Log.Infof("Running pre-install script: %s", script.Path)
+	for _, sysConfig := range config.SystemConfigs {
+		for _, script := range sysConfig.PreInstallScripts {
+			ReportActionf("Running pre-install script: %s", path.Base(script.Path))
+			logger.Log.Infof("Running pre-install script: %s", script.Path)
 
-		err = shell.ExecuteLive(squashErrors, "chmod", "a+x", script.Path)
-		if err != nil {
-			return
-		}
+			shell.ExecuteLive(squashErrors, "chmod", "a+x", script.Path)
```

Comment:
Why do we no longer check error here?

Reply:
During the iso installation process, the scripts will be executed within a read-only file system and thus running in this command will actually throw an error and break the installation. Since preinstall script would only be used for kickstart-style unattended iso installation and not for normal image builds, I will remove this line as it's not needed

Reply:
Could you please leave a comment explaining that?